evalml/model_understanding/_partial_dependence_fast_mode_utils.py

```diff
@@ -0,0 +1,103 @@
+import pandas as pd
+
+from evalml.utils.woodwork_utils import infer_feature_types
+
+
+def _get_cloned_feature_pipelines(
+    features,
+    pipeline,
+    variable_has_features_passed_to_estimator,
+    X_train,
+    y_train,
+):
+    """Clones and fits pipelines for partial dependence fast mode.
+
+    Args:
+        pipeline (PipelineBase or subclass): Fitted pipeline that will be cloned.
+        features (string, tuple[int or string]): The target feature for which to create the partial dependence plot for.
+            If features is a string, it must be a valid column name in X.
+            If features is a tuple of int/strings, it must contain valid column integers/names in X.
+        variable_has_features_passed_to_estimator (dict[str, bool]): Dictionary mapping variable names to whether or not
+            the variable has an impact on predictions. Variables that do not have features passed to the estimator
+            will not have an impact on predictions and therefore do not need to have pipelines fitted for them.
+        X_train (pd.DataFrame, np.ndarray): The data that was used to train the original pipeline. Will
+            be used to train the cloned pipelines.
+        y_train (pd.Series, np.ndarray): The target data that was used to train the original pipeline. Will
+            be used to train the cloned pipelines.
+
+    Returns:
+        dict[str, PipelineBase or subclass]: Dictionary mapping feature name to the pipeline pipeline
+            fit for it.
+    """
+    if X_train is None or y_train is None:
+        raise ValueError("Training data is required for partial dependence fast mode.")
+
+    X_train = infer_feature_types(X_train)
+
+    # Make sure that only components that are capable of handling fast mode are in the pipeline
+    new_parameters = pipeline.parameters
+    for component in pipeline.component_graph.component_instances.values():
+        new_parameters = component._handle_partial_dependence_fast_mode(
+            X_train,
+            new_parameters,
+        )
+
+    # Create a fit pipeline for each feature
+    cloned_feature_pipelines = {}
+    for variable in features:
```

Comment:
Very good call. I also think it should just be `(list(str))`, because it's getting passed in from `_partial_dependence_calculation`, which already says it'll just be a list of strings at that point.